src/Analyzers/Orchestration/OrchestrationAnalyzer.cs

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.DurableTask.Analyzers.Orchestration;
+
+public abstract class OrchestrationAnalyzer : DiagnosticAnalyzer
+{
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            var knownSymbols = new KnownTypeSymbols(context.Compilation);
+
+            if (knownSymbols.FunctionOrchestrationAttribute == null || knownSymbols.FunctionNameAttribute == null ||
```

Comment:
Can you use pattern matching here as well?

not sure if this syntax is correct:

`if (knownSymbols is { FunctionOrchestrationAttribute is null or FunctionNameAttribute is null ... })`

Reply:
I could make it work by:

```cs
if (knownSymbols is
    { FunctionOrchestrationAttribute: null } or { FunctionNameAttribute: null } or
    { TaskOrchestratorInterface: null } or { TaskOrchestratorBaseClass: null } or
    { DurableTaskRegistry: null })
{
    // symbols not available in this compilation, skip analysis
    return;
}
```

Giving the additional `} {` tokens per check, which syntax do you prefer? I am fine with either.

>  I experiment a bit with the syntax and code generated in [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABAzAAgwJhwBEcBvAXwFgAoG7PQgYTJpzeIH4cEcBeHAHYBXADYiA3K3ZEuATz6DREqWxV5cGAIxIcASwEAXHAFlNACgCULau1s4AbgEMoOMAoEwA7jkaXJNuzZdADMzMAA6Hl5+YTEcNDRXcPloxTELNUC2DAB2HAAGfyy8PJR8IvYqANVqnAB6OpwYAEchXScRGEM1ei0dfSNjfEtrYqcXNxivHz9M9hCwvQBnMgQQNJFyHGgyWXXYzYza4tyCisDTsvOcKts1BpwAExDgmFhu2t7tPUMTLBHSHM2ONXO5pr4LNdbAs3LoVqQ1hsADQ4PYbchWB4QAQieQhHCOOIABygECJK2cMA2BKMBgAFlSlo4ALZUgy6VlAuynQpcy7lNRVKpAA=).

Reply:
I am good with either. Another option is to just replace `== null` with `is null` in the existing `||` style.

Reply:
Ok, I will keep at it is for now.